### ISSUE TITLE
[Feature request] Auto Save

### ISSUE THREAD

**Issue by ShobanChiddarth** [JasonStein/Notepads#739]
**Auto Save**
We need auto save like VS Code

**Alternatives**
Vs code has that autosave option so we no need to press CTRL+S every time

**Comment by soumyamahunt** [comment]
You can use session snapshot and don't have to worry about saving your files.

**Comment by arpitkaushal** [comment]
> You can use session snapshot and don't have to worry about saving your files.

However, that would mean that if I close the notepads without saving the file, and open that file in another editor, it'll show the last saved state of that file, instead of also showing the edits that were made in the Notepads app.

So, it does make more sense to add autosave instead.

**Comment by soumyamahunt** [comment]
> However, that would mean that if I close the notepads without saving the file, and open that file in another editor, it'll show the last saved state of that file, instead of also showing the edits that were made in the Notepads app.

In that case all the edits you have done before closing Notepads will be saved in session-snapshot and will be loaded next time you start the app.

**Comment by arpitkaushal** [comment]
I apologize if it wasn't clear in my last message. 

Contents of `helloworld.md` file:
```
Hi, this is a file.
```
1. I open helloworld.md in Notepads.
2. I add a line `Hello from Notepads`. So, the contents of file is now: `Hi, this is a file.  Hello from Notepads.`
3. I close Notepads.
4. I open helloworld.md in VS Code.
5. I see `Hi, this is a file.` instead of `Hi, this is a file.  Hello from Notepads.`
6. But if I open Notepads, then I see the edits I made in Step 2.

So, it'll be better to have autosave in which case I'd have seen the edits even in Step 5.

I hope I am clear now.

**Comment by raunak111** [comment]
Yes I agree, autosave is quite an essential feature and I feel its not much harder to implement either.
